preprocess/bom_test.go

```diff
@@ -58,3 +58,17 @@ func TestRemoveBOM(t *testing.T) {
 		require.True(t, bytes.HasPrefix(n, []byte("#!")), "%s does not start with shebang: %#v", fn, n)
 	}
 }
+
+func Test_encodeToUTF8(t *testing.T) {
```

Comment:
@boumenot in case I'd say "best effort" instead of forgiving. if we cannot determine the encoding to be utf8 or utf16, we just preserve the original encoding.